crates/router/src/core/user.rs

```diff
@@ -1821,30 +1821,23 @@ pub async fn send_verification_mail(
 #[cfg(feature = "recon")]
 pub async fn verify_token(
     state: SessionState,
-    req: auth::ReconUser,
+    user: auth::UserFromToken,
 ) -> UserResponse<user_api::VerifyTokenResponse> {
-    let user = state
+    let user_in_db = state
         .global_store
-        .find_user_by_id(&req.user_id)
+        .find_user_by_id(&user.user_id)
         .await
-        .map_err(|e| {
-            if e.current_context().is_db_not_found() {
-                e.change_context(UserErrors::UserNotFound)
-            } else {
-                e.change_context(UserErrors::InternalServerError)
-            }
+        .change_context(UserErrors::InternalServerError)
+        .attach_printable_lazy(|| {
+            format!(
+                "Failed to fetch the user from DB for user_id - {}",
+                user.user_id
+            )
         })?;
```

Comment:
NIT: `UserFromToken` already has a method to get user from db.

Reply:
Will take this separately 👍